src/plugin/pid/pid.cpp

```diff
@@ -36,11 +36,12 @@ using namespace dmtcp;
 extern "C" pid_t dmtcp_update_ppid();
 
 static string pidMapFile;
-map<pthread_mutex_t*, pid_t>& mapMutexVirtTid()
+dmtcp::map<pthread_mutex_t*, pid_t>& mapMutexVirtTid()
 {
-  static map<pthread_mutex_t*, pid_t> *instance = NULL;
+  static dmtcp::map<pthread_mutex_t*, pid_t> *instance = NULL;
   if (instance == NULL) {
-    instance = new map<pthread_mutex_t*, pid_t>();
+    void *buffer = JALLOC_MALLOC(1024*1024);
+    instance = new (buffer) dmtcp::map<pthread_mutex_t*, pid_t>();
```

Comment:
> Is this equal to ... ?

No, the two are not equal.

Placement new forces the object to be allocated in the pre-allocated region and calls the constructor. See https://isocpp.org/wiki/faq/dtors#placement-new